### DISCUSSION TITLE
Multiple input sources (RTSP or physical)

### DISCUSSION THREAD

**Discussion by MatthewBCooke** [mcguirepr89/BirdNET-Pi#774] [Ideas]
I was wondering if there's a way to use multiple input sources for analysis. I currently am running an RTSP stream to BirdNET-Pi and its working great. I would love the ability to add multiple audio source and either 1. combine them into a stereo track for analysis, or 2. analyze both audio streams separately. 

Has this been looked into? Does anyone know of a way to combine RTSP streams into a stereo track. 

My concerns for the stereo option is if there is a difference in latency between the microphones, things might get pretty wonky for analysis. 

Let me know your thoughts, and whether or not this may or may not be possible. Right now I'm running multiple instances of BirdNET to account for this but it would be awesome to combine them into one interface.

**Comment by ehpersonal38** [comment]
A secret little feature of BirdNET-Pi is that you can actually analyze as many RTSP streams as you want concurrently. All you have to do is separate your RTSP stream URLs by commas when you're inputting them on the Advanced Settings page. Here's an example with 3 RTSP streams:

![chrome_KaxgM6ppmJ](https://user-images.githubusercontent.com/103586016/222613971-73eafad9-43ff-48e8-84b2-115e69f55e05.png)

The only catch is that your Pi _may_ not be able to support the workload of recording multiple streams at once. So you'll have to do some tests to see if it's stable or not. I recommend running the tool `htop` to analyze your CPU/Memory usage.

**Reply by MatthewBCooke** [reply]
I can't thank you enough!

**Reply by ampsu** [reply]
I wonder. It is possible to have multiple RTSP streams but is it possible to seperate the detections? If I got two streams and all the detections are mixed up, I cant possibly where the bird was :D

**Reply by jaredb7** [reply]
While the database isn't setup to record that information it is possible to figure out the source from the filename since they contain _RTST**X**_ where X is 1 for the first stream 2 for 2nd, etc etc.
Ideally we'd add an extra column to the database to record the source and populate that when the detection is written

In theory we have something like "Source: <rtsp hostname/ip>" on the detections page, maybe also have it on the most recent detection too.
I'll put it on the to-do list.
![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/799998/87b6328d-377a-4ae9-8340-56727a5fa00c)
